### PR TITLE
[EDGORDERS-83-IT]. Add tls integration test, add test scope BC deps, …

### PR DESCRIPTION
**Pull request by BKadirkhodjaev**
TLS integration test using BC keystore provider

### CODE REVIEW

File: src/test/java/org/folio/edge/core/EdgeVerticleTlsIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider;
+import org.folio.edge.core.utils.SslConfigurationUtil;
+import org.folio.edge.core.utils.test.TestUtils;
+import org.junit.*;
+import org.junit.runner.RunWith;
```
Comment (azizbekxm) [inline src/test/java/org/folio/edge/core/EdgeVerticleTlsIntegrationTest.java:19]: first line don't need I think

File: pom.xml [hunk 2]
```diff
@@ -89,6 +92,25 @@
       <version>5.11.0</version>
       <scope>test</scope>
     </dependency>
+    <!-- Library provides the basic cryptographic functionality complying with FIPS for testing -->
+    <dependency>
+      <groupId>org.bouncycastle</groupId>
+      <artifactId>bc-fips</artifactId>
+      <version>${bc-fips.version}</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (SerhiiNosko) [inline pom.xml:101]: for our simple test it is enought only bc-fips dependency

File: src/test/java/org/folio/edge/core/EdgeVerticleTlsIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      .listen(config.getInteger(Constants.SYS_PORT), http -> logger.info("Server started on port {}", config.getInteger(Constants.SYS_PORT)));
+
+    final OkapiClientFactory okapiClientFactory = OkapiClientFactoryInitializer.createInstance(vertx, config);
+    final OkapiClient okapiClient = okapiClientFactory.getOkapiClient(TENANT);
+    final WebClientOptions okapiWebClientOptions = (WebClientOptions) FieldUtils.readDeclaredField(okapiClient.client, "options", true);
+
+    assertTrue(okapiWebClientOptions.isSsl());
+    assertNotNull(okapiWebClientOptions.getTrustOptions());
+
+    final WebClientOptions clientOptions = new WebClientOptions();
```
Comment (SerhiiNosko) [inline src/test/java/org/folio/edge/core/EdgeVerticleTlsIntegrationTest.java:81]: we already have declared okapiWebClientOptions

File: src/test/java/org/folio/edge/core/EdgeVerticleTlsIntegrationTest.java (hunk 1 above)
Comment (BKadirkhodjaev) [reply src/test/java/org/folio/edge/core/EdgeVerticleTlsIntegrationTest.java:23]: Do you mean remove I need to remove `import org.junit.*;`?

**Review by SerhiiNosko — APPROVED** [review]

File: src/test/java/org/folio/edge/core/EdgeVerticleTlsIntegrationTest.java (hunk 1 above)
Comment (azizbekxm) [reply src/test/java/org/folio/edge/core/EdgeVerticleTlsIntegrationTest.java:25]: need use specific imports instead of implicit import style

**Review by azizbekxm — APPROVED** [review]

**Review by TarasSpashchenko — APPROVED** [review]

**Review by steveellis — APPROVED** [review]

**Review by TarasSpashchenko — APPROVED** [review]

**Pull request merged**